src/addon/addon_manager.cpp

```diff
@@ -522,12 +542,25 @@ void
 AddonManager::mount_old_addons()
 {
   std::string mountpoint;
-  for (auto& addon : m_installed_addons) {
-    if (is_old_enabled_addon(addon)) {
-      if (PHYSFS_mount(addon->get_install_filename().c_str(), mountpoint.c_str(), 1) == 0)
+  for (auto& addon : m_installed_addons) 
+    {
+      if (is_old_enabled_addon(addon)) 
       {
-        log_warning << "Could not add " << addon->get_install_filename() << " to search path: "
-                    << PHYSFS_getLastErrorCode() << std::endl;
+        if (g_config->use_local_path) 
+        {
+                if (PHYSFS_mount(addon->get_install_filename().c_str(), mountpoint.c_str(), 0) == 0)
+          {
+              log_warning << "Could not add " << addon->get_install_filename() << " to search path: "
+              << PHYSFS_getLastErrorCode() << std::endl;
+          }
+        }
+        else
+        {
+        if (PHYSFS_mount(addon->get_install_filename().c_str(), mountpoint.c_str(), 1) == 0)
+        {
+                          log_warning << "Could not add " << addon->get_install_filename() << " to search path: "
+                          << PHYSFS_getLastErrorCode() << std::endl;
+        }
       }
     }
   }
```

Comment:
Indentation fix here to, there's an extra +2 tabs _before_ an open bracket after the `for` and the open bracket before the last `if` doesn't add +2 spaces. Also, some lines have way too many spaces :)